92/felix.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Definition for singly-linked list.
+ * class ListNode {
+ *     val: number
+ *     next: ListNode | null
+ *     constructor(val?: number, next?: ListNode | null) {
+ *         this.val = (val===undefined ? 0 : val)
+ *         this.next = (next===undefined ? null : next)
+ *     }
+ * }
+ */
+
+function reverseBetween(head: ListNode | null, left: number, right: number): ListNode | null {
+    let valArr: { [index: string]: number } = {};
+    
+    function rec(node: ListNode | null, index) {
+        if (!node) return;
+        // record nodes that needs to be reversed
+        if (index >= left && index <= right) {
+            valArr[index] = node.val;
```

Comment:
I think we can use a queue to push values in, then pop it out on the other side, sort like a FIFO.

Reply:
make sense, but there is no queue built-in in js. And as far as memory consumption, there is no difference between saving values into an object and a so called queue (an array in js).

Reply:
haha, sure, it's just makes me feel better.

Reply:
haha, sure, it's just makes me feel better.